### PR TITLE
DESK-238

### PR DESCRIPTION
**Pull request by JamieRuderman**
Was trying to re-connect to maintain online state when a service was offline. So now added online state to connection so that we don't try to connect if offline. Also allowed to update state in Tray menu and react appropriately.

### CODE REVIEW

File: backend/src/ElectronApp.ts [hunk 1]
```diff
@@ -125,7 +125,6 @@ export default class ElectronApp {
   }
 
   private openWindow = (openDevTools?: boolean) => {
-    console.log('open window')
```
Comment (bstrech) [inline backend/src/ElectronApp.ts]: Do you really want to keep this console.log

File: backend/src/TrayMenu.ts [hunk 2]
```diff
@@ -85,15 +90,19 @@ export default class TrayMenu {
         const location = hostName(connection, this.privateIP)
         result.push({
           label: connection.name,
-          icon: connection.active ? iconConnected : iconOnline,
+          icon: connection.active ? iconConnected : connection.online ? iconOnline : iconOffline,
```
Comment (bstrech) [inline backend/src/TrayMenu.ts]: Ternary in a ternary.  Is that a pattern we want to keep?

Comment (JamieRuderman) [reply backend/src/TrayMenu.ts]: I'm okay with it.  :)

File: backend/src/ElectronApp.ts (hunk 1 above)
Comment (JamieRuderman) [reply backend/src/ElectronApp.ts]: No - it's been removed in another pr: https://github.com/remoteit/desktop/compare/DESK-315

**Pull request merged**